src/util.ts

```diff
@@ -130,3 +130,48 @@ export function joinAll(a: ArrayBuffer[]): ArrayBuffer {
 
     return buffer;
 }
+
+// Consumes data:
+//   PrivateToken challenge="abc...", token-key="123...",
+//   PrivateToken challenge="def...", token-key="234...",
+//   PrivateToken challenge=ghi..., token-key=345...
+// Parse WWW-Authenticate according to RFC9110 Section 11.6.1 https://www.rfc-editor.org/rfc/rfc9110#section-11.6.1
+export function parseWWWAuthenticate(header: string): string[] {
+    const ALPHA = 'A-Za-z';
+    const DIGIT = '0-9';
+    const tokenChar = `!#$%&'*+\\-\\.^_\`|~${DIGIT}${ALPHA}`;
+    const tchar = '[' + tokenChar + ']';
+    const OWS = '[ \\t]*';
+    const RWS = '[ \\t]+';
+    const BWS = OWS;
+    const qdtext = '[\\t\\x21\\x23-\\x5B\\x5D-\\x7E\\x80-\\xFF]';
+    const quotedPair = '\\\\([ \\t' + tokenChar + '\\x21-\\x7E\\x80-\\xFF])';
```

Comment:
Not sure what you were trying to define here, but the space and tabs are reversed, and `tokenChar` does not belong there:
```suggestion
    const quotedPair = /\\[\t \x21-\x7e\x80-\xff]/.source;
```

Based on the grammar:
```
quoted-pair = "\" ( HTAB / SP / VCHAR / obs-text )
```
From https://www.rfc-editor.org/rfc/rfc5234.html
```
         VCHAR          =  %x21-7E
                                ; visible (printing) characters
```

Reply:
updated